enhancements/maintenance/machine-maintenance-operator.md

```diff
@@ -0,0 +1,241 @@
+
+A POC of the idea can be found [here](https://github.com/dofinn/machine-maintenance-operator).
+
+## Motivation
+
+Currently the machine-api fails to accurately detect the state of machines (validated in AWS) when they have been terminated/stopped by the cloud provider. A provider may terminate/stop and instance when there is an associated maintenance scheduled for it. This is also the case for users manually terminating/stopping machines via the console (validated in AWS). The MMO is a proactive approach to first detecting maintenances and/or machine state from the cloud provider, then executing the required to enable the machine-api to continue managing machines and machinesets. This operator would work in conjunction with MachineHealthCheck implimentation. 
+
+### Goals
+
+List the specific goals of the proposal. How will we know that this has succeeded?
+The machine-maintenance-operator will:
+* detect maintenances and execute them manually at their earliest and most convenient time
+* validate the state of the machine against its machine CR and the cloud providers API
+* validate the state of the machine behind a cloud providers load balancers and provide an alert upon recognition of a miss-match of state
+
+### Non-Goals
+
+This is not a solution for managing maintenances or machine state for master machine roles. 
+
+## Proposal
+
+### User Stories [optional]
+
+#### Story 1
+As a customer using OKD/OCP/OSD, I want my cluster/s to be proactive in handling events that are initiated by the cloud provider hosting my cluster. This will assure me that OKD/OCP/OSD accounts for machines in the cluster that require terminating or stopping to be proactively maintained and addressed with intention. 
+
+#### Story 2
+As an SRE supporting OSD, I want automated detection of cloud provider scheduled maintenance events. This will enable automated processes to proactively handle the event and execute the required with intent and validation, while also alerting me if required. 
+
+As an SRE supporting OSD, I want automated validation of machine state as advertised by the machine-api against the cloud provider. This will enable automated processes to proactively handle the event and execute the required with intent and validation, while also alerting me if required. 
+
+As an SRE supporting OSD, I want automated validation of machine state as advertised by the machine-api against the cloud providers load balancers (health checks). This will enable automated processes to proactively handle the event and execute the required with intent and validation, while also alerting me if required. 
```

Comment:
It's not clear what load balancers really have to do with machine state.  ELBs have health checks.  If a machine fails health checks, it's removed from ELBs.

For other types of load balancers (network ingress?), those are handled by operators, and again, health checks.

Reply:
This operator would alert on (potentially remediate) the removal of a node from an ELB on specific conditions. For example a failed health check.

Reply:
I guess I need to know more about how instances are ending up behind the ELBs in the first place.  I thought the network ingress operator took care of this.